src/solve.c

```diff
@@ -27,8 +27,78 @@
  */
 int solve(UNMOLESTED_INPUT *unin, MOLESTED_INPUT *in)
 {
-  LOG("SOLVE CALLED", 2);
+  LOG("solve CALLED", 2);
+  int toReturn = process_and(unin, in);
+  LOG("solve RETURNING", 2);
+  return toReturn;
+}
+
+int process_or(int* values, int* data, int datalen)
+{
+  LOG("process_or CALLED", 2);
+  int i;
+  for(i = 0; i < datalen; i++)
+  {
+    int val = data[i];
+    if(val < 0)
+    {
+      val = -val;
+      val = !values[val-1];
+    }else
+    {
+      val = values[val-1];
+    }
+    if(val)
+    {
+      LOG("process_or RETURNING",2);
+      return 1;
+    }
+  }
+  LOG("process_or RETURNING",2);
+  return 0;
+}
+
+int process_and(UNMOLESTED_INPUT *unin, MOLESTED_INPUT *in)
+{
+  int i, j;
+  //int input[unin->nbvars];
+  int * input = (int*)malloc(sizeof(int)* unin->nbvars);
 
-  LOG("SOLVE RETURNING", 2);
+  
+  for(i = 0; i < (1<< unin->nbvars); i++)
```

Comment:
Pull the pointer dereferencing into a statement before the loop so it only happens once.